File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/util/scheduler/ImmediateSchedulerProvider.java
package com.elveselimoski.starterapp.util.scheduler;

import androidx.annotation.NonNull;
import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

import javax.inject.Inject;

/**
 * Created by Elvedin Selimoski on 11/29/17. mail: elveselimoski@gmail.com
 */

public class ImmediateSchedulerProvider implements SchedulerProvider {

    @Inject
    ImmediateSchedulerProvider() {
    }

    @NonNull
    @Override
    public Scheduler computation() {
        return Schedulers.trampoline();
    }

    @NonNull
    @Override
    public Scheduler io() {
        return Schedulers.trampoline();
    }

    @NonNull
    @Override
    public Scheduler ui() {
        return Schedulers.trampoline();
    }
}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/util/scheduler/ImmediateSchedulerProvider.kt
package com.elveselimoski.starterapp.util.scheduler

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

/**
 * Created by Elvedin Selimoski on 11/29/17. mail: elveselimoski@gmail.com
 */

class ImmediateSchedulerProvider : SchedulerProvider {

    override fun computation(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun io(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun ui(): Scheduler {
        return Schedulers.trampoline()
    }
}
